### PR TITLE
Added favorites feature for containers.

### PR DESCRIPTION
**Pull request by danielamram**
Hi, tried to resolve the issue #3067 ,
Some screenshots: 

**favorite container -**
favorites1

**favorites containers and hover on container -**
favorites2

I'm not sure if the code containerActions.js file is right, 
have a problem that i want to save the favorite names in the localstorage any toggle,
Because if the computer shutdown and the app exiting, i want the favorite to be saved before,
Review it and comment me if something can be improved.

Thank you,

- Added favorites feature for containers.
- Fixed side-bar width css, added cursor: pointer to containers.

Signed-off-by: Daniel Amram <danielamram0@gmail.com>

### CODE REVIEW

File: src/components/Containers.react.js [hunk 1]
```diff
@@ -32,17 +32,23 @@ var Containers = React.createClass({
 
   sorted: function (containers) {
     return _.values(containers).sort(function (a, b) {
-      if (a.State.Downloading && !b.State.Downloading) {
-        return -1;
-      } else if (!a.State.Downloading && b.State.Downloading) {
-        return 1;
+      if (a.Favorite && !b.Favorite) {
+        return -1
```
Comment (FrenchBen) [inline src/components/Containers.react.js]: missing `;`

File: src/components/Containers.react.js [hunk 2]
```diff
@@ -32,17 +32,23 @@ var Containers = React.createClass({
...
+      } else if (!a.Favorite && b.Favorite) {
+        return 1
```
Comment (FrenchBen) [inline src/components/Containers.react.js]: missing `;`

**Comment by danielamram** [comment]
@FrenchBen Do you mean that the name of the container isn't unique? Its preffered to use the id to indicate the favorites?

**Comment by FrenchBen** [comment]
ID may be better yes, as the container name isn't unique - Kitematic keeps using the name of the image, which doesn't make it unique.

**Comment by danielamram** [comment]
Got it, but i saw that all the app using the name as an unique id - the container reducer, the services and more it seems weird to use the id where whole the app using name...
More than that, there's a function 'generateContainerName' that generate an unique id.

**Comment by FrenchBen** [comment]
may be ok - didn't get a chance to try it out with a few different containers. If we merge and issues come up, can I assign them to you?

**Comment by danielamram** [comment]
@FrenchBen For sure ,Thank you.
What's preffered to be done? issues or the roadmap?
I want to continue contribute this repo...

**Comment by FrenchBen** [comment]
there are a couple of feature requests and issues that are open - Feel free to start knocking some of these out.

**Comment by FrenchBen** [comment]
Hi @danielamram - While playing around with 1 container and a fresh install of Docker, I ran into the following error:
`Uncaught TypeError: Cannot read property 'includes' of null`
Full stack trace:
```js
Uncaught TypeError: Cannot read property 'includes' of null
    at AltAction.toggleFavorite (/Users/Test/kitematic/build/actions/ContainerActions.js:87:20)
    at Constructor.toggleFavoriteContainer (/Users/Test/kitematic/build/components/ContainerListItem.react.js:40:42)
    at Object.executeDispatch (/Users/Test/kitematic/build/node_modules/react/lib/EventPluginUtils.js:110:21)
    at executeDispatch (/Users/Test/kitematic/build/node_modules/react/lib/SimpleEventPlugin.js:305:40)
    at forEachEventDispatch (/Users/Test/kitematic/build/node_modules/react/lib/EventPluginUtils.js:95:7)
    at Object.executeDispatchesInOrder (/Users/Test/kitematic/build/node_modules/react/lib/EventPluginUtils.js:119:3)
    at executeDispatchesAndRelease (/Users/Test/kitematic/build/node_modules/react/lib/EventPluginHub.js:46:22)
    at Array.forEach (native)
    at forEachAccumulated (/Users/Test/kitematic/build/node_modules/react/lib/forEachAccumulated.js:23:9)
    at Object.processEventQueue (/Users/Test/kitematic/build/node_modules/react/lib/EventPluginHub.js:253:5)
    at runEventQueueInBatch (/Users/Test/kitematic/build/node_modules/react/lib/ReactEventEmitterMixin.js:18:18)
    at Object.handleTopLevel [as _handleTopLevel] (/Users/Test/kitematic/build/node_modules/react/lib/ReactEventEmitterMixin.js:44:5)
    at handleTopLevelImpl (/Users/Test/kitematic/build/node_modules/react/lib/ReactEventListener.js:80:24)
    at ReactDefaultBatchingStrategyTransaction.perform (/Users/Test/kitematic/build/node_modules/react/lib/Transaction.js:134:20)
    at Object.batchedUpdates (/Users/Test/kitematic/build/node_modules/react/lib/ReactDefaultBatchingStrategy.js:66:19)
    at Object.batchedUpdates (/Users/Test/kitematic/build/node_modules/react/lib/ReactUpdates.js:110:20)
```

Could you see what caused this? Steps to reproduce:
1. Start kitematic
2. Create a new container using the `hello-world-nginx` image
3. Stop the container
4. Click on the ❤️

**Comment by danielamram** [comment]
Ill check it out...

**Comment by danielamram** [comment]
Hi @FrenchBen I know how to fix the issue, just to init the array with empty array.
Do you want that ill open a new pull request? or how i'm adding another pull request to this one?

**Comment by FrenchBen** [comment]
@danielamram if you could open up the PR that would be great! Thanks

**Comment by danielamram** [comment]
Opened a pull request with the change, its just 1 line - #3541

**Pull request merged**